features/netsocket/cellular/generic_modem_driver/PPPCellularInterface.cpp

```diff
@@ -560,13 +570,23 @@ nsapi_error_t PPPCellularInterface::connect()
     bool did_init = false;
     const char *apn_config = NULL;
 
-    if (dev_info.ppp_connection_up) {
+    if (_connect_status == NSAPI_STATUS_GLOBAL_UP || _connect_status == NSAPI_STATUS_LOCAL_UP) {
```

Comment:
As you're touching it, this would be a good time to make it `if (is_connected())`.